tests/python_functional/src/driver_io/network/network_io.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2015-2018 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+from pathlib2 import Path
+
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.common.random_id import get_unique_id
+from src.driver_io.file.file import File
+from src.helpers.loggen.loggen import Loggen
+
+
+class NetworkIO():
+    def __init__(self, ip, port, transport):
+        self.__ip = ip
+        self.__port = port
+        self.__transport = transport.replace("_", "-").replace("'", "").replace('"', "")
+
+    def write(self, content):
+        loggen_input_file_path = Path(tc_parameters.WORKING_DIR, "loggen_input_{}.txt".format(get_unique_id()))
+        loggen_input_file = File(loggen_input_file_path)
```

Comment:
Instead of using File class we can import FileIO class which has it's own write() method. 
If you missing out something (file object close() or something like that) from that API we should fix it there
I would strive to using already existing API's as many places as we can.

Reply:
I think `IO` classes are on the same level of the hierarchy, I'd rather not make any dependencies between them.

Reply:
Ohh, understand. So you mean if we modify because of something (specific driver feature) the read() / write() methods of the IO layers, we do not wanted to broke simple read() / write() methods in other places which are not related to syslog-ng Drivers. This is true. According to this later we should remove the FileIO dependency from syslog_ng_config.py and console_log_reader.py and snmptrapd/conftest.py files